### PR TITLE
Added hethom genotype viz

### PR DESCRIPTION
**Pull request by jpdna**
to show progress, todo:
- clear saved data from notebook
- add test

note: the hethom function does not extend HistogramDistribution, perhaps it could, but I am not certain until I look at it more.  My version here uses dataframe/SQL rather than rdd.

Let me know if prefer me to try further to use HistogramDistribution, or if a version of this and CallRate is OK for now not using it.

### CODE REVIEW

**Comment by akmorrow13** [comment]
@jpdna  ADAM is ready to cut a release soon it would be great to get this in. Any updates on the TODOs?

**Comment by jpdna** [comment]
> Any updates on the TODOs?

The above commit addresses the remaining todos.   Please let me know any comments/changes.

File: mango-python/bdgenomics/mango/genotypes.py [hunk 1]
```diff
@@ -105,3 +134,117 @@ def __init__(self, ss, genotypeRDD, sample = 1.0):
...
+           matplotlib axis to plot and computed data
+        """
+
+        if(not testMode):
+            figsize = kwargs.get('figsize',(10,5))
+            fig, ax = plt.subplots(figsize=figsize)
+            hist = ax.hist(self.callRates,bins=100)
+            return ax, self.callRates
+        else:
+            return None, self.callRates
```
Comment (akmorrow13) [inline mango-python/bdgenomics/mango/genotypes.py]: add newline

File: mango-python/bdgenomics/mango/test/genotypes_test.py [hunk 2]
```diff
@@ -48,6 +48,24 @@ def test_VariantsPerSampleDistributionSampling(self):
...
+        expected = [5.0, 0.6, 0.14285714285714285, 0.16666666666666666, 1.6666666666666667]
+        assert( expected == data)
+
+    def test_GenotypeCallRatesDistribution(self):
+        ac = ADAMContext(self.ss)
+        testFile = self.resourceFile("genodata.v3.test.vcf")
+
+        genotypes = ac.loadGenotypes(testFile)
+        _, data =  GenotypeCallRatesDistribution(self.ss, genotypes, sample=1.0).plot(testMode= True)
+        expected = [0.9505208333333334, 0.8776041666666666, 0.8880208333333334, 0.9401041666666666, 0.9348958333333334, 0.9010416666666666]
```
Comment (akmorrow13) [inline mango-python/bdgenomics/mango/test/genotypes_test.py]: It is not a good idea to test like this. Results can change based on precision. Do a sum, count, first or something with rounding.

**Comment by jpdna** [comment]
Hi @akmorrow13  looking 
https://amplab.cs.berkeley.edu/jenkins/job/mango-prb/797/HADOOP_VERSION=2.7.5,SCALAVER=2.11,SPARK_VERSION=2.2.2,label=ubuntu/consoleText
containing below, I am not clear how the changes I made to the genotype tests (to round the values) would have caused this error, when the previous commit did pass for me locally.
I'll try later today to move my changes over onto current master so that I'm starting on top of a version I know passes tests.


```
==================================== ERRORS ====================================
_________ ERROR collecting bdgenomics/mango/pileup/test/test_embed.py __________
bdgenomics/mango/pileup/test/test_embed.py:29: in <module>
    from ipywidgets.embed import embed_data, embed_snippet, embed_minimal_html, dependency_state
E   ImportError: No module named embed
=================================== FAILURES ===================================
__________________________ MangoVizTest.test_features __________________________

self = <test_application.MangoVizTest testMethod=test_features>

    def test_features(self):
        track = pileup.Track(viz="features", label="myFeatures", source=pileup.sources.GA4GHFeatureJson('{}'))
    
>       x = pileup.PileupViewer(locus="chr17:1-250", reference="hg19", tracks=[track])
```

**Comment by jpdna** [comment]
Replaced by #469, which rebases on master.
Problems compiling locally were fixed with updated Spark to 2.3.2,  #469 Jenkins test are still running but should pass....

**Pull request closed without merging**